File: RoboHome.Models/Flip.cs
using System;
namespace RoboHome.Models
{
    public class Flip
    {
        public int Id { get; set; }
        public SwitchState Direction { get; set; }
        public int Hour { get; set; }
        public int Minute { get; set; }
        public TimeOfDay TimeOfDay { get; set; }
        public Time Time { get; set; }

        //For EF
        public Flip(int id, SwitchState direction, int hour, int minute, TimeOfDay tod) {
            this.Id = id;
            this.Direction = direction;
            this.Hour = hour;
            this.Minute = minute;
            this.TimeOfDay = tod;
        }
    }
}

File: RoboHome.Site/wwwroot/ts/models/flip.ts
import {SwitchState, Time} from '../models';
export class Flip {
    constructor(public id: number = -1,
                public direction: SwitchState = SwitchState.Off,
                public time: Time = new Time()) {}
    static fromJson(json: any): Flip {
        if (!json) return new Flip();
        return new Flip(
            json.id,
            json.direction,
            Time.fromJson(json.time),
        )
    }
}